Kernel/Arch/riscv64/mcontext.h

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright (c) 2023, Sönke Holz <sholz8530@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <Kernel/API/POSIX/sys/types.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+struct __attribute__((packed)) __mcontext {
+    uintptr_t x[31];
+    uintptr_t sp;
+    uintptr_t pc;
```

Comment:
See comment above; if the file is written like this it *could* be shared between RV32 and RV64, but since we're disregarding 32-bit anyways, that's not necessary.